pkg/cmd/pulumi/new_test.go

```diff
@@ -745,9 +745,6 @@ func TestSetFail(t *testing.T) {
 		{
 			Array: []string{`my:name[-1]=value`},
 		},
-		{
-			Array: []string{`my:name[1]=value`},
```

Comment:
Why doesn't this case error anymore?

Reply:
I'm pretty sure the original evaluator incorrectly treated this as an error--it ought to be possible to index into top-level array-valued properties.